src/plugins/intel_cpu/src/nodes/inverse.cpp

```diff
@@ -0,0 +1,236 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "inverse.hpp"
+
+#include "nodes/common/cpu_memcpy.h"
+#include "openvino/core/parallel.hpp"
+#include "openvino/core/type.hpp"
+#include "openvino/op/constant.hpp"
+#include "openvino/op/inverse.hpp"
+#include "utils/bfloat16.hpp"
+
+// Parallel LU decomposition algorithm with partial pivoting
+// Based on the lectures by Prof. Dr. Thomas Huckle, Parallel Numerics
+
+namespace ov {
+namespace intel_cpu {
+namespace node {
+
+Inverse::Inverse(const std::shared_ptr<ov::Node>& op, const GraphContext::CPtr& context)
+    : Node(op, context, NgraphShapeInferFactory(op, EMPTY_PORT_MASK)) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        THROW_CPU_NODE_ERR(errorMessage);
+    }
+
+    auto inverse_op = as_type_ptr<op::v14::Inverse>(op);
+    m_adjoint = inverse_op->get_adjoint();
+
+    constant = ConstantType::StrictNoConst;
+
+    m_const_input = is_type<op::v0::Constant>(op->get_input_node_ptr(INPUT_PORT));
+}
+
+bool Inverse::isSupportedOperation(const std::shared_ptr<const ov::Node>& op, std::string& errorMessage) noexcept {
```

Comment:
Please validate that the rank is static here due to the plugin does not support dynamic ranks.

Reply:
There are common checks for dynamic rank in the base class (CPU Node) constructor.
Isn't that enough?
https://github.com/openvinotoolkit/openvino/blob/3238290df5ecb9e0eb0c4254d20bf668c3bc1408/src/plugins/intel_cpu/src/node.cpp#L64-L70

https://github.com/openvinotoolkit/openvino/blob/3238290df5ecb9e0eb0c4254d20bf668c3bc1408/src/plugins/intel_cpu/src/node.cpp#L84-L91